<launch>

	<!-- Launch defenders nodes-->
	<!-- defender1 -->
	<group ns="defender1">

		<arg name="id" default="0"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="13" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
    </node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
    </node>

	</group>

	<!-- defender2 -->
	<group ns="defender2">

		<arg name="id" default="1"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="15" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
    </node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
    </node>

	</group>

	<!-- defender3 -->
	<group ns="defender3">

		<arg name="id" default="2"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="17" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
    </node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
    </node>
	</group>

	<!-- defender4 -->
	<group ns="defender4">

		<arg name="id" default="3"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="21" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
    </node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
    </node>
	</group>

	<!-- defender5 -->
	<group ns="defender5">

		<arg name="id" default="4"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="23" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
		</node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
		</node>
	</group>

	<!-- defender6 -->
	<group ns="defender6">

		<arg name="id" default="5"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="28" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
		</node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
		</node>
	</group>

	<!-- defender7 -->
	<group ns="defender7">

		<arg name="id" default="6"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="30" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
		</node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
		</node>
	</group>

	<!-- defender8 -->
	<group ns="defender8">

		<arg name="id" default="7"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="66" />

		<node pkg="dlp" type="dlp_node" name="dlp_node_$(arg id)"  required="true" output="screen">
		</node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="defender_dynamics_node_$(arg id)"  required="true" output="screen">
		</node>
	</group>

	<!-- Launch attackers nodes-->
	<!-- attacker1 -->
	<group ns="attacker1">

		<arg name="id" default="0"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="91" />
		<param name="isATTACKER" value="True" />
		<param name="alpha" value="0.8" />
		<param name="beta" value="-0.2" />
		<param name="cmd_step" value="0.1" />

		<node pkg="dlp" type="dlp_node_attacking" name="dlp_attacker_node_$(arg id)"  required="true" output="screen">
    </node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="attacker_dynamics_node_$(arg id)"  required="true" output="screen">
    </node>
	</group>

	<!-- attacker2 -->
	<group ns="attacker2">

		<arg name="id" default="1"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="99" />
		<param name="isATTACKER" value="True" />
		<param name="alpha" value="0.8" />
		<param name="beta" value="-0.2" />
		<param name="cmd_step" value="0.1" />

		<node pkg="dlp" type="dlp_node_attacking" name="dlp_attacker_node_$(arg id)"  required="true" output="screen">
    </node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="attacker_dynamics_node_$(arg id)"  required="true" output="screen">
    </node>
	</group>

	<!-- attacker3 -->
	<group ns="attacker3">

		<arg name="id" default="2"/>
		<rosparam command="load" file="$(find dlp)/config/problem_setup_2Dsimulation.yaml" />
		<param name="myID" type="int" value="$(arg id)" />
		<param name="home_sector" type="int" value="85" />
		<param name="isATTACKER" value="True" />
		<param name="alpha" value="0.8" />
		<param name="beta" value="-0.2" />
		<param name="cmd_step" value="0.1" />

		<node pkg="dlp" type="dlp_node_attacking" name="dlp_attacker_node_$(arg id)"  required="true" output="screen">
		</node>
		<node pkg="dlp" type="agent_dynamics_node.py" name="attacker_dynamics_node_$(arg id)"  required="true" output="screen">
		</node>
	</group>


</launch>
